### ISSUE TITLE
App with embedded jackd

### ISSUE THREAD

**Issue by kltdwrds** [jackaudio/jack2#672]
Is there a "right way" to compile / distribute a binary linked to a jackd bundled w/ the distribution?  There are a lot of jack client's in the OSS world, and i've never seen this done.   

I realize competing jack servers is bad news, but my application requires a very specific jack build, and the user base is very non-technical - I'd like them to download and run my application without having to require an additional download/install for jack.

**Comment by falkTX** [comment]
you lose the benefits of using jack that way, so why not just talk to the hw directly then?
using something like rtaudio, portaudio, sdl or other libraries that can handle that.

**Comment by kmatheussen** [comment]
On Fri, Nov 20, 2020 at 3:04 AM Kyle T. Edwards
<notifications@github.com> wrote:
>
> Is there a "right way" to compile / distribute a binary linked to a jackd bundled w/ the distribution? There are a lot of jack client's in the OSS world, and i've never seen this done.
>
> I realize competing jack servers is bad news, but my application requires a very specific jack build, and the user base is very non-technical - I'd like them to download and run my application without having to require an additional download/install for jack.
>

You could look at how Radium does this, although it's not very nice code.
1. Radium uses a modified version of weakjack:
https://github.com/kmatheussen/radium/tree/master/weakjack
2. Radium includes a modified version jack:
https://github.com/jackaudio/jack2/compare/develop...kmatheussen:mingw.jack_local
3. Radium includes a modified version of qjackctl:
https://github.com/jackaudio/jack2/compare/develop...kmatheussen:mingw.jack_local

**Comment by kmatheussen** [comment]
>
> You could look at how Radium does this, although it's not very nice code.
> 1. Radium uses a modified version of weakjack:
> https://github.com/kmatheussen/radium/tree/master/weakjack
> 2. Radium includes a modified version jack:
> https://github.com/jackaudio/jack2/compare/develop...kmatheussen:mingw.jack_local
> 3. Radium includes a modified version of qjackctl:
> https://github.com/jackaudio/jack2/compare/develop...kmatheussen:mingw.jack_local

Link 3 should be https://github.com/kmatheussen/qjackctl

**Comment by kmatheussen** [comment]
Actually it's not very hard. Just use the included version of jack and
qjackctl binaries that is included with the radium binaries (the demos
are fine), and compile your program with the modified version of
weakjack instead of the official weakjack.

**Comment by kltdwrds** [comment]
> you lose the benefits of using jack that way, so why not just talk to the hw directly then?
> using something like rtaudio, portaudio, sdl or other libraries that can handle that.

Yes I agree it's not ideal - I'm dependent on an open source project that's dependent on jack, and I want to bundle jack via a custom build of the OS jack client so that users of my app don't need to install jack separately.
